src/vm/typehashingalgorithms.h

```diff
@@ -0,0 +1,90 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+// ---------------------------------------------------------------------------
+// Generic functions to compute the hashcode value of types
+// ---------------------------------------------------------------------------
+
+#pragma once
+#include <stdlib.h>
+
+// DIFFERENT FROM CORERT: Hashing code currently different from CoreRT
+
+//
+// Returns the hashcode value of the 'src' string
+//
+inline static DWORD ComputeNameHashCode(LPCUTF8 src)
```

Comment:
Existing code was using a variety of types to store hash code: `ReadyToRunInfo::TryLookupTypeTokenFromName` used DWORD, some other code used unsigned int, while some used int. I tried to standardize on a single type in all the code involved in this feature, but looks like DWORD was a bad choice. I'll switch to int.